src/main/cs/Infinispan/HotRod/Config/Configuration.cs

```diff
@@ -11,11 +11,12 @@ public class Configuration
     {
         private Infinispan.HotRod.SWIG.Configuration config;
         private IMarshaller marshaller;
-
-        internal Configuration(Infinispan.HotRod.SWIG.Configuration config, IMarshaller marshaller)
+        private object authCallback;
```

Comment:
Since authCallback is completely unrelated to the top-level configuration object, maybe it would be better to hold a reference to AuthenticationConfiguration here? AuthConf has the reference to the callback so it should not disappear. WDYT?

Reply:
OK, I've added a reference to AuthenticationConfiguration. The Security->Authentication configuration is kept only on the C side at the moment.